### DISCUSSION TITLE
Different flavors of CoDa regression

### DISCUSSION THREAD

**Discussion by johannesbjork** [mikemc/differential-abundance-theory#18] [Q&A]
Hi Mike,

This is a great resource/reading material. Thanks for making it public! 

I was wondering, I have you looked into linear models on clr transformed relative abundances vs multinomial regression suggested by Morton et al. (2019)?

I think the former is becoming common place as it's more straightforward to run. But the latter has the benefit that ranks of coefficients are identical on both relative and absolute data. I use Justin Silverman's fido package to run it, but it would be useful if there also were a frequentist way of running it. 

Thanks
Johannes

**Comment by mikemc** [comment]
Hi @johannesbjork, thanks for your interest and question. Here are some thoughts - 

I do indeed think that analyses based on fold changes in ratios such as those performed by Justin Silverman's fido package are a good way to account for taxonomic bias when the interest is in analyzing relative abundances. In our 2019 eLife paper, we pointed out that such analyses are in theory robust to taxonomic bias and not just the 'compositional bias'  discussed by Morton et al (driven by changes in total reads / total abundance). Since this result has been misunderstood and is still not widely known, we emphasize it more in this current paper.

However changes in ratios (or the CLR) can be hard to interpret and don't directly correspond to changes in absolute abundance, which is often what researchers are interested in. Therefore a major goal of this present manuscript, and distinction from Morton et al, is that we're considering ways that true changes in absolute abundance can be measured, not just in ratios. We also emphasize the problems with proportion-based analyses because 1) they are still extremely common despite the work of Morton and Silverman et al and 2) a  common approach to measuring absolute abundance currently is to multiply the proportions by a measure of total density, which as we show may not address taxonomic bias even if it addresses 'compositional bias'.

There are other tools implementing the multinomial logistic normal model that is used by fido; Amy Willis's DivNet package is one, but I'm not sure if it is possible to perform a differential abundance analysis with it. Greg Gloor's Aldex2 package provides a semi-frequentist ad-hoc approximation approach, specifically for differential abundance with CLR or other LR transform. (Though the notion of 'differential abundance' in the compositional framework is a bit fuzzy, since fundamentally it is the change in the ratios of species that is being considered.)
